### PR TITLE
Cluster version added to labels in manifests

### PR DESCRIPTION
**Pull request by tushar-tushar**

### CODE REVIEW

File: commons/src/main/java/io/hyscale/commons/models/ClusterVersionInfo.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.commons.models;
+
+import io.hyscale.commons.constants.ToolConstants;
+
+public class ClusterVersionInfo {
+
+    private String majorVersion;
```
Comment (SameeraGrandhi) [inline commons/src/main/java/io/hyscale/commons/models/ClusterVersionInfo.java:22]: Good one !
can we just refer major & minor...version is inferred from the model name I feel.

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/processor/ClusterVersionProvider.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    
+    @Autowired
+    private K8sClientProvider clientProvider;
+    
+    @Autowired
+    private K8sVersionHandler k8sVersionHandler;
+    
+    private ClusterVersionInfo version;
+    
+    public ClusterVersionInfo getVersion(AuthConfig authConfig) throws HyscaleException {
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/processor/ClusterVersionProvider.java:41]: Since it is dealing with K8s resources it can consume K8sAuthorisation

File: commons/src/main/java/io/hyscale/commons/utils/ResourceLabelBuilder.java [hunk 3]
```diff
@@ -70,6 +70,18 @@
 
 		return labels;
 	}
+	
+    public static Map<ResourceLabelKey, String> setVersion(Map<ResourceLabelKey, String> labels, String version) {
```
Comment (SameeraGrandhi) [inline commons/src/main/java/io/hyscale/commons/utils/ResourceLabelBuilder.java:74]: Why is this required ?

Comment (tushar-tushar) [reply commons/src/main/java/io/hyscale/commons/utils/ResourceLabelBuilder.java:74]: While setting the resource label map in manifest context under ManifestContextBuilder.

**Comment by tushar-tushar** [comment]
Making changeds to add labels in Agents, will update the PR accordingly.

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/KubernetesVolumeUtil.java [hunk 4]
```diff
@@ -90,6 +90,10 @@ public static String getVolumeName(V1PersistentVolumeClaim pvc) {
 		}
 		String pvcName = pvc.getMetadata().getName();
 		Map<String, String> labels = pvc.getMetadata().getLabels();
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/KubernetesVolumeUtil.java:92]: Why ? 
It's not compatible with version < 1.15 right !

File: manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/VolumeTemplatesHandler.java [hunk 5]
```diff
@@ -82,29 +81,32 @@
         serviceMetadata.setAppName(manifestContext.getAppName());
 
         List<ManifestSnippet> snippetList = new ArrayList<>();
-
+        Map<String, String> customLabels = manifestContext
```
Comment (SameeraGrandhi) [inline manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/VolumeTemplatesHandler.java:84]: We can make this is a first-class field in the manifest context?
Gives more visibility to users. What do you say ?

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/KubernetesVolumeUtil.java (hunk 4 above)
Comment (tushar-tushar) [reply deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/KubernetesVolumeUtil.java:92]: This will work for versions 1.15 and above, and for other versions, we still have the fallback part.

File: manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/VolumeTemplatesHandler.java (hunk 5 above)
Comment (tushar-tushar) [reply manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/VolumeTemplatesHandler.java:84]: I agree, will make those changes.
